### ISSUE TITLE
New Suspense changes broke solid-app-router redirecting

### ISSUE THREAD

**Issue by minht11** [solidjs/solid#668]
**Describe the bug**
I have a page which redirects from "/" -> "/home/target". After upgrading solid from 1.1.0 to 1.1.5 I observed that opening page directly in '/' showed me blank results, no route component would be displayed, URL did update correctly though. Removing `<Suspense />` would fix things. I can consistently reproduce this behavior locally.

I tried to slim down the reproduction (code below) and I can see it happening every time when running that code snippet in vite dev mode, but not with vite production or CodeSandbox, note that when running my full app code issue shows up regardless of mode. This is likely very timing sensitive.

**To Reproduce**
```tsx
import { Navigate, Router, useRoutes } from 'solid-app-router'
import { render } from 'solid-js/web'
import { Component, createSignal, Show, Suspense } from 'solid-js'

export const ROUTES = [
  {
    path: '/home/target',
    component: () => <div>Home</div>,
  },
  {
    path: '/',
    children: [{ path: '/' }, { path: '/home' }],
    component: () => <Navigate href='/home/target' />,
  },
]

const AppContent = () => {
  const Routes = useRoutes(ROUTES)

  return (
    <div>
      <Suspense>
        <Routes />
      </Suspense>
    </div>
  )
}

const LoadAsyncStateProvider: Component = (props) => {
  const [isVisible, setIsVisible] = createSignal(false)

  setTimeout(() => {
    setIsVisible(true)
  }, 100)

  return <Show when={isVisible()}>{props.children}</Show>
}

const App = () => (
  <Router>
    <LoadAsyncStateProvider>
      <AppContent />
    </LoadAsyncStateProvider>
  </Router>
)

render(() => <App />, document.body)
```
1. Load code snippet locally in SPA mode.
2. Go to "/", so that redirection code runs.
2. Observe that URL does change, but nothing is displayed.

**Expected behavior**
Component is displayed correctly.

**Additional context**
If I remove `<LoadAsyncStateProvider />` component is displayed correctly, but URL doesn't change. Everything works fine with Solid version 1.1.0.

**Comment by ryansolid** [comment]
Thanking for taking time to reproduce. There were some changes in 1.1.1 that appear to be missing some edge cases. This has been my focus the last week to just improve this whole mechanism but having a good set of use cases here to base tests on will help a lot.

**Comment by minht11** [comment]
After a bit of testing I encountered another issue which is present in both 1.1.0 and 1.1.5 versions. After a redirect any store values accessed below `<LoadAsyncStateProvider />` wouldn't be reactive. 

```tsx
const LoadAsyncStateProvider: Component = (props) => {
  const [isVisible, setIsVisible] = createSignal(false)
  setTimeout(() => setIsVisible(true), 100)

  return <Show when={isVisible()}>{props.children}</Show>
}

const App = () => (
  <StoreProvider>
    <LoadAsyncStateProvider>
      <div>
        {/* Removing suspense fixes things */}
        <Suspense>
          <Routes />
        </Suspense>
        {/* Store values accessed here would become not reactive */}
        <ComponentWhichUsesStore />
      </div>
    </LoadAsyncStateProvider>
  </StoreProvider>
)
```

Thankfully I can [reproduce](https://codesandbox.io/s/zen-framework-sm9se?file=/index.js) it inside CodeSandbox.
Change URL to "/" and see that value doesn't change.
If page is loaded directly to "/home/target" value changes as expected.

**Comment by minht11** [comment]
I just noticed but with version Solid version 1.1.6 browser back forward buttons do not work at all.

**Comment by ryansolid** [comment]
Yeah router was banking on some guarentees that are no longer true. Looking at a fix.

**Comment by ryansolid** [comment]
Hmm.. the browser button stuff is codesandbox specific. It works fine outside. I see the same behavior before 1.1.6. In any case 0.1.9 of the router should fix the issue you were seeing.
